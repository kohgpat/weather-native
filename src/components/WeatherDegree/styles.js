import styled from "styled-components/native";

export const Degree = styled.Text``;
