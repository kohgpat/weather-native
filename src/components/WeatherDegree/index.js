import React from "react";
import PropTypes from "prop-types";
import * as styles from "./styles";

const WeatherDegree = ({ units }) => {
  if (!units) {
    return null;
  }

  if (units === "celcius") {
    return <styles.Degree>°C</styles.Degree>;
  }

  if (units === "fahrenheit") {
    return <styles.Degree>°F</styles.Degree>;
  }

  return null;
};

WeatherDegree.defaultProps = {
  units: null
};

WeatherDegree.propTypes = {
  units: PropTypes.string.isRequired
};

export default WeatherDegree;
