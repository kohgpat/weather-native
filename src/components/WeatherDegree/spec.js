/*global it, expect*/
import React from "react";
import WeatherDegree from "./index";
import { shallow, mount } from "enzyme";

it("renders null", () => {
  const weatherDegree = shallow(<WeatherDegree units="" />);
  expect(weatherDegree).toMatchSnapshot();
});

it("renders Celcius", () => {
  const weatherDegree = shallow(<WeatherDegree units="celcius" />);
  expect(weatherDegree).toMatchSnapshot();
});

it("renders Fahrenheit", () => {
  const weatherDegree = shallow(<WeatherDegree units="fahrenheit" />);
  expect(weatherDegree).toMatchSnapshot();
});

it("returns null", () => {
  const weatherDegree = shallow(<WeatherDegree units="" />);
  expect(weatherDegree.type()).toEqual(null);
});

it("renders Celcius symbol", () => {
  const weatherDegree = mount(<WeatherDegree units="celcius" />);
  expect(weatherDegree.text()).toContain("C");
});

it("renders Fahrenheit symbol", () => {
  const weatherDegree = mount(<WeatherDegree units="fahrenheit" />);
  expect(weatherDegree.text()).toContain("F");
});
