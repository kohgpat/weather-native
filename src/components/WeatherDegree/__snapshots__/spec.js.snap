// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Celcius 1`] = `
<styled.span>
  °C
</styled.span>
`;

exports[`renders Fahrenheit 1`] = `
<styled.span>
  °F
</styled.span>
`;

exports[`renders null 1`] = `""`;
